### PR TITLE
fac-build: init at 0.5.3

### PR DESCRIPTION
**Pull request by dpercy**
###### Motivation for this change

Add fac build system: http://sites.science.oregonstate.edu/~roundyd/fac/

The Nix package name "fac" is taken, so does "fac-build" make sense instead?

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
       (fac only supports Linux today.)
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/tools/build-managers/fac/default.nix [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+      Fac is a general-purpose build system inspired by make that utilizes
+      ptrace to ensure that all dependences are enumerated and that all
+      source files are added to a (git) repo. An important feature of fac
+      is that it automatically handles dependencies, rather than either
+      complaining about them or giving an incorrect build. Currently, fac
+      only runs on linux systems, but on those systems it is incredibly
+      easy to use!
+    '';
+    homepage = "https://physics.oregonstate.edu/~roundyd/fac";
+    license = [ licenses.gpl2 ];
```
Comment (danieldk) [inline pkgs/development/tools/build-managers/fac/default.nix:48]:
```suggestion
    license = licenses.gpl2Plus;
```

Should not be in a list. Also, `gpl2` is deprecated and should be `gpl2Only` or `gpl2Plus`. The readme states

> Fac is free software, and is licensed under the GNU General Public License, version 2 or later.

So, `gpl2Plus` is the correct license.

File: pkgs/development/tools/build-managers/fac/default.nix [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    platforms = platforms.linux;
+    maintainers = maintainers.dpercy;
```
Comment (danieldk) [inline pkgs/development/tools/build-managers/fac/default.nix:50]:
```suggestion
    maintainers = [ maintainers.dpercy ];
```

File: pkgs/development/tools/build-managers/fac/default.nix [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+
+rustPlatform.buildRustPackage rec {
+  name = "fac-build";
+  version = "0.5.3";
+
+  src = fetchFromGitHub {
+    owner = "droundy";
+    repo = "fac";
+    rev = "05c8507e3e4c5484a6f0e1393cea0c37d1d679ae";
+    sha256 = "sha256:1gifrlb31jy8633rnhny58ccp3wlmd338129c6sh0h1a38vkmsxk";
```
Comment (danieldk) [inline pkgs/development/tools/build-managers/fac/default.nix:11]:
```suggestion
    sha256 = "1gifrlb31jy8633rnhny58ccp3wlmd338129c6sh0h1a38vkmsxk";
```

More canonical.

File: pkgs/development/tools/build-managers/fac/default.nix [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+    # fac includes a unit test called ls_files_works which assumes it's
+    # running in a git repo. Nix's sandbox runs cargo build outside git,
+    # so this test won't work.
+    ./skip-git-ls-files-test.patch
+  ];
+
+  # workaround for missing Cargo.lock file
+  cargoPatches = [ ./cargo-lock.patch ];
+
+  cargoSha256 = "sha256:1kk1d957h34z0z5xvzdrbjz99qddw20kpkrw7pn5nxnanyjavln5";
```
Comment (danieldk) [inline pkgs/development/tools/build-managers/fac/default.nix:24]:
```suggestion
  cargoSha256 = "1kk1d957h34z0z5xvzdrbjz99qddw20kpkrw7pn5nxnanyjavln5";
```

**Review by danieldk — CHANGES_REQUESTED** [review]
Thank you for your first contribution :+1: 

I have added some requested changes.

File: pkgs/development/tools/build-managers/fac/default.nix [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+
+  buildInputs = with pkgs; [ git ];
```
[the remark this answers is not in the document]
Comment (dpercy) [reply pkgs/development/tools/build-managers/fac/default.nix]: Oh nice, I didn't know about that distinction.

File: pkgs/development/tools/build-managers/fac/default.nix [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+
+  # fac calls git at runtime, expecting it to be in the PATH,
+  # so we need to patch it to call git by absolute path instead.
+  postPatch = ''
+    sed -i "/std::process::Command::new/ s:git:${pkgs.git}/bin/git:" src/git.rs
```
[the remark this answers is not in the document]
Comment (dpercy) [reply pkgs/development/tools/build-managers/fac/default.nix]: Nice, I guess this is a NixOS-specific helper?

**Comment by dpercy** [comment]
Thanks for the feedback! I think I've addressed everything so far.

**Review by danieldk — APPROVED** [review]
LGTM. Thanks a lot!

Result of `nixpkgs-review pr 97868` [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>fac-build</li>
  </ul>
</details>

**Comment by dpercy** [comment]
Thanks!

**Pull request merged**